resources/schemas/dbscripts/sqlserver/snd-0.00-18.20.sql

```diff
@@ -71,7 +71,7 @@ CREATE TABLE snd.Pkgs (
 
    CONSTRAINT PK_SND_PKGS PRIMARY KEY (PkgId),
    CONSTRAINT FK_SND_PKGS_CONTAINER FOREIGN KEY (Container) REFERENCES core.Containers (EntityId),
-   CONSTRAINT FK_SND_PKGS_QCSTATE FOREIGN Key (QcState) REFERENCES core.QCState (RowId)
+   CONSTRAINT FK_SND_PKGS_QCSTATE FOREIGN Key (QcState) REFERENCES core.DataStates (RowId)
```

Comment:
What about the servers that already have these `FK_SND_PKGS_QCSTATE` and `FK_SND_EVENTS_QCSTATE` constaints? Do they need a separate SQL upgrade script to drop and recreate them with the new reference?

Reply:
Grr.  Yes, they probably do, which surely means this all needs to be a deferred upgrade script, which makes me ever so happy.

Reply:
This might be a case where we should reference SND (or other module's schemas) in the core upgrade script and do a drop if exists on the FK. SND would then create a new version of the FK in its upgrade.

Reply:
That's essentially what I was thinking of for the deferred upgrade. If we're ok orchestrating this without deferring by core exhibiting knowledge of other modules, that would be somewhat preferable.

Reply:
Yes, I think the small bit of encapsulation leaking OK given the problems that deferred upgrades can create

Reply:
Well, turns out that, surprisingly, both supported databases do the right thing with these foreign key constraints after the table rename, so I think we're good to go.